pkgs/servers/pingvin-share/backend.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildNpmPackage
+, vips
+, pkg-config
+, nodePackages
+, src
+, version
+}:
+
+let
+  prisma = nodePackages.prisma;
+in
+
+buildNpmPackage {
+  pname = "pingvin-share-backend";
+  inherit version;
+
+  src = "${src}/backend";
+
+  postPatch = ''
+    cp ${./backend-package-lock.json} package-lock.json
```

Comment:
Why is this necessary? I've been running my own package of pingvin-share successfully without a custom lock file.

Reply:
During my tests I ran into the error `Error: getaddrinfo EAI_AGAIN github.com` which has been fixed with with the custom lockfiles (patched with `npm-lockfile-fix`)

Reply:
Hm, is this still an issue? I can't reproduce this on `aarch64-darwin`/`aarch64-linux` with:

```nix
# backend
npmDepsHash = "sha256-rXksLrT/WCy7ESEBNeKes+CxBdjmV6uZO24Rkf1urbk=";

# frontend
npmDepsHash = "sha256-C0YKbiZzdj8HIGM3ivQ9+KhGsC5IfVz54ZhKZmlXcoA=";
```

But I've also seen some packages run `npm-lockfile-fix` in `postFetch` of `fetchFromGitHub`.

Reply:
I re-ran my tests and indeed this part does not seem necessary so I delete the lockfiles and the `postPatch` phases